field_float_nullable/fields.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2025 ACSONE SA/NV
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+from odoo import fields
+
+
+class FloatNullable(fields.Float):
+    """
+    The FloatNullable field is a custom Odoo field type that extends the standard fields.Float
+    to allow NULL values in the database column.
+    By default, Odoo uses 0.0 as the default value for fields.Float, which can lead
+    to ambiguity between "unset" and "zero" values.
+    """
+
+    type = "float_nullable"
+    column_type = ("numeric", "numeric")
+    search = True
+
+    def convert_to_column(self, value, record, values=None, validate=True):
+        if value is None or value == "" or (value is False and type(value) is bool):
+            return None
+        try:
+            return float(value) if value is not None else None
+        except (ValueError, TypeError):
+            return None
```

Comment:
Is it not better to call `super().convert_to_column()` instead of re-implementing the conversion logic?